packages/test/test-utils/src/testObjectProvider.ts

```diff
@@ -122,14 +122,63 @@ function getDocumentIdStrategy(type?: TestDriverTypes): IDocumentIdStrategy {
     }
 }
 
+export class ErrorTrackingLogger extends TelemetryLogger{
```

Comment:
Seems like there's a path to merging this and `MockLogger`, something to think about as we tune this.